src/tests/ftest/util/osa_utils.py

```diff
@@ -310,6 +310,31 @@ def set_cont_class_properties(self, oclass="S1"):
             prop = prop.replace("rf:1", rf_value)
         self.container.properties.value = prop
 
+    def assert_on_exception(self, out_queue):
+        """Assert on exception while executing an application.
+
+        Args:
+            out_queue (queue): Check whether the queue is
+            empty. If empty, app (ior, mdtest) didn't encounter error.
+        """
+        if self.out_queue.empty():
+            pass
+        else:
+            exc = out_queue.get(block=False)
+            out_queue.put(exc)
+            raise exc
```

Comment:
Maybe I just don't have context, but I'm not sure why you need to use a queue for this. You're only handling a single item here. What if there are multiple items to raise? Also not sure why you put it back because I assume you don't want to raise the same failure in the future call to this method.

Reply:
Good question...  The reason why the data is put back in the queue is while handing Main Program ---> ThreadA (Parent)--> ThreadB (child)---> TestMethod. The exception raised in TrestMethod is catched by ThreadB but never passed to ThreadA... 

I am using this technique on online tests (it is similar to fork). You fork the threads (for IOR to run in parallel) and perform OSA tasks on the main program.

Having a queue is the only way, main program can see the exceptions generated in the child threads.

Reply:
Okay. Thanks.